### ISSUE TITLE
Add socket activation support for systemd

### ISSUE THREAD

**Issue by davidstrauss** [PowerDNS/pdns#956]
Reasons it's awesome:
- Better dependency management at boot time. Especially for pdns-recursor, many services may need to connect to run properly. Socket activation allows systemd to open the socket in early boot, allowing services depending on pdns or pdns-recursor to start in parallel.
- More secure isolation. Nothing needs to start as root because systemd handles the initial listening on a low port and passes it into the daemon. With network namespaces, daemons can run in containers isolating them from unnecessary network access, in or out. Passed-in sockets still work.
- On-demand service start-up, if desired.
- Restarting and reconfiguring the service don't interrupt listening.
- More consistent listener configuration relative to other services using socket activation.

Some C-based examples are here: http://0pointer.de/blog/projects/socket-activation.html

**Comment by bjoe2k4** [comment]
Any update on this? Using the recursor is quite a pain if other services depend on it.

**Comment by cmouse** [comment]
Can't you workaround with After=pdns-recursor.service?

**Comment by bjoe2k4** [comment]
Yes, this is what i ended up doing (actually After=nss-lookup.target), but you have to do it for *every* service that requires dns lookups on startup. Still, socket activation would be nice to have.
